pages/secure/sec_graylog_beats.rst

```diff
@@ -0,0 +1,102 @@
+
+
+Deploy and configure
+====================
+
+Graylog
+-------
+
+HTTPS
+^^^^^
+
+Place the ``.key`` and ``.crt file`` to your Graylog server in the configuration dir (/etc/graylog/server/) and add them to the Graylog server.conf. In addition change the ``rest_listen_uri`` and ``web_listen_uri`` to **https**. You might need to cover other settings in a multinode cluster or special setups - just read the comments of the settings inside of the server.conf.
+
+When using the collector-sidecar, use the **https** URI in the ``sidecar_configuration.yml``.
+
+After restart of Graylog the web interface and the API is served via https only. No automatically redirect from http to https is made.
+
+TLS Beats Input
+^^^^^^^^^^^^^^^
+
+To enable TLS on the input one certificate - the certificate file and the private key file - that can be used for the input. We recommend to use the same certificate that is already used to secure the webinterface. Just reference the files `TLS cert file` and `TLS private key file` in the Beats Input configuration and restart the input. 
+
+The ingesting client will verify the presented certificate against his know CA certificates, if that is successfull communication will be establised using TLS. 
+
+
+Add client authentication to beats input
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create one directory (``/etc/graylog/server/trusted_clients``) that will hold all client certificates you allow to connect to the beats input - that must be available on all Graylog server that have the input enabled. Write that path in the beats input configuration `TLS Client Auth Trusted Certs` and select **required** for the option `TLS client authentication`.
+
+After this setting is saved only clients that provide a certificate that is trusted by the CA and is placed inside the configured directory (``/etc/graylog/server/trusted_clients``) can deliver messages to Graylog.
+
+Beats
+-----
+
+Stock
+^^^^^
+
+When using the stock beat that is `provided by elastic <https://www.elastic.co/downloads/beats>`__ configure a `logstash output <https://www.elastic.co/guide/en/beats/filebeat/6.x/logstash-output.html#logstash-output>`__. The SSL configuration can be found as the second point in the `description by elastic <https://www.elastic.co/guide/en/beats/filebeat/6.x/configuring-ssl-logstash.html>`__ . This is::
+
+	output.logstash:
+  		hosts: ["logs.mycompany.com:5044"]
+  		ssl.certificate_authorities: ["/etc/ca.pem"]
+  		ssl.certificate: "/etc/client.crt"
+  		ssl.key: "/etc/client.key"
+
+
+Place your previous create certificates on the server where you installed beats and adjust the configuration to your needs.
+
+The certificate (``.crt``) file of the beat need to be placed at the Graylog server in the configured directory for trusted clients, only if you have enabled that feature at the beats input in Graylog and want client authentication.
```

Comment:
The certificate (``.crt``) file of beats needs to be placed on the Graylog server in the configured directory for trusted clients only if you have enabled that feature at the beats input in Graylog and want client authentication.